aws/aws-parallelcluster
# Add Tags at compute resource level

### Description of changes
Add support for Tags section under compute resource level.
```
  SlurmQueues:
    - Name: compute
      AllocationStrategy: lowest-price
      ComputeResources:
        - Name: defaultcompute
          Instances:
            - InstanceType: c5.xlarge
          MinCount: 0
          MaxCount: 505
          Tags:
            - Key: mytag1
              Value: tagvalue13
```
The tags section under queue level support queue update strategy
* Add validator to make sure no duplicate tags within the same section
* Add warning to warn compute resource tags will override queue tags, queue tags will override cluster level tags when there are duplicated tag keys.

### Tests
* Manually test
  * Test cluster creation with Tags section in compute resources, 
    * verify duplicated tags in the Compute Resource section override Queue level
    * verify duplicated tags in the Compute Resource section override Cluster level
  * Test cluster update with Tags section in compute resources
  * Test cluster creation/update is blocked when there are duplicated Tags within the same section
* Unit test

### References
* Link to related PRs in other packages (i.e. cookbook, node).
* Link to documentation useful to understand the changes.

### Checklist

Please review the [guidelines for contributing](../CONTRIBUTING.md) and [Pull Request Instructions](https://github.com/aws/aws-parallelcluster/wiki/Git-Pull-Request-Instructions).

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
